src/Serilog.Sinks.Map/MapLoggerConfigurationExtensions.cs

```diff
@@ -21,6 +21,15 @@
 namespace Serilog
 {
     /// <summary>
+    /// A function delegate to select a key value given a log event, if exists.
+    /// </summary>
+    /// <typeparam name="TKey">The type of the key value.</typeparam>
+    /// <param name="logEvent">The log event.</param>
+    /// <param name="key">The selected key.</param>
+    /// <returns>true, if a key can be selected, or false, otherwise.</returns>
+    public delegate bool KeySelector<TKey>(LogEvent logEvent, out TKey key);
```

Comment:
Because this is in namespace `Serilog`, it will be brought into scope whenever `ILogger`, `Log` and so-on are. This might create an annoying IntelliSense clash with other `Key*` types, like `KeyValuePair<,>`.

Perhaps we move it under `Serilog.Sinks.Map`? The name doesn't have to be in scope in order to use the lambda signature anyway, so shouldn't impact usage.

Reply:
Good point, will do that.